src/data_scraper/scraper_binance.py

```diff
@@ -12,77 +12,87 @@
 class BinanceScraper:
     def __init__(self, currency_to_buy=config.CURRENCY_TO_BUY, currency_to_sell=config.CURRENCY_TO_SELL,
                  all_currencies=config.ALL_CURRENCIES, interval=config.INTERVAL, dev_run=True, **kwargs):
-        os.makedirs(f'{config.FOLDER_TO_SAVE}/binance', exist_ok=True)
         self.name = "Binance"
         self.dev_run = dev_run
         self.client = BinanceClient(key=credentials.BINANCE_API_KEY, secret=credentials.BINANCE_API_SECRET)
         self.interval = interval
         self.currency_to_buy = currency_to_buy
         self.currency_to_sell = currency_to_sell
-        all_currencies.remove(self.currency_to_buy)
-        self.currency_pairs = [f'{self.currency_to_buy}{self.currency_to_sell}',
-                               *[f'{currency}{self.currency_to_sell}' for currency in all_currencies]]
-        self.col_names = ['Open time', 'Open', 'High', 'Low', 'Close', 'Volume', 'Close time', 'Quote asset volume',
+        self.currency_pairs = [
+            f'{self.currency_to_buy}{self.currency_to_sell}',
+            *[f'{currency}{self.currency_to_sell}' for currency in all_currencies if currency != self.currency_to_buy]
+        ]
+        self.col_names = ['Timestamp (ms)', 'Open', 'High', 'Low', 'Close', 'Volume', 'Close time', 'Quote asset volume',
                           'Number of trades', 'Taker buy base asset volume', 'Taker buy quote asset volume', 'Ignore']
-
-        self.dataset_path = f"{config.FOLDER_TO_SAVE}/binance"
-        self.dataset_name = f"{self.currency_to_buy}{self.currency_to_sell}"
+        self.datatypes = [int, float, float, float, float, float, int, float, int, float, float, int]
+        self.dataset_path = f"{config.FOLDER_TO_SAVE}/{self.name}/{self.interval}"
+        self.cache_data = None
 
     def get_data(self, start_time, end_time):
-        """Get Binance API exchange rates for all selected currencies. Operates in UTC timezone.
-        By default, it will try to continue from the latest available data point if the last saved timestamp is ahead
-        of the timestamp specified in the config, to save time instead of loading already available data."""
-
+        """
+        Load the exchange data from Binance. Always scrapes data in production.
+        Tries to load data from disk in development. If it is not possible, scrapes the data and saves in parquet.
+        """
+        if not self.dev_run:  # Always scrape if in production
```

Comment:
I feel like this class shouldn't know if it's a dev run or not, since we're already doing it splitting the DataService in two. (Training and Production). What I would do is that TrainingDataDervice should call `self.get_stored_data` and ProductionDataService `self.scrape_data`. This way we avoid specifying twice that it's a dev_run (once when selecting the data service and a second time by specifying the dev_run flag value).

Reply:
Makes sense, removed